### PR TITLE
Configure pagination parameters

### PR DESCRIPTION
**Pull request by giorgiosironi**
For https://github.com/elifesciences/issues/issues/4478

### CODE REVIEW

**Comment by giorgiosironi** [comment]
I'm asserting the current behavior at the moment, looking into the body response of 404s.

File: app/digests/tests/test_digest_api.py [hunk 1]
```diff
@@ -134,3 +135,43 @@ def test_can_filter_on_digest_stage(stage: str,
...
+
+    last_page = 4
+    total = sum([_count_items(page) for page in range(1, last_page)]) + _count_items(page=last_page)
+
+    assert total == len(multiple_published_digests)
+
+@pytest.mark.django_db
+def test_invalid_page(client: Client):
+    response = client.get(f'{DIGESTS_URL}?page=0')
+    assert response.status_code == 404
```
Comment (thewilkybarkid) [inline app/digests/tests/test_digest_api.py]: I'd expect a `400`.

File: app/digests/tests/test_digest_api.py [hunk 2]
```diff
@@ -134,3 +135,43 @@ def test_can_filter_on_digest_stage(stage: str,
...
+
+@pytest.mark.django_db
+def test_invalid_per_page_is_ignored(client: Client, multiple_published_digests):
+    response = client.get(f'{DIGESTS_URL}?per-page=0')
+    assert response.status_code == 200
```
Comment (thewilkybarkid) [inline app/digests/tests/test_digest_api.py]: I'd expect a `400`.

File: app/digests/tests/test_digest_api.py [hunk 3]
```diff
@@ -134,3 +135,43 @@ def test_can_filter_on_digest_stage(stage: str,
...
+def test_page_too_large(client: Client, multiple_published_digests):
+    all_digests = len(multiple_published_digests)
+    response = client.get(f'{DIGESTS_URL}?per-page={all_digests}&page=2')
+    assert response.status_code == 404
+
+@patch('digests.pagination.DigestPagination.max_page_size', 5)
+@pytest.mark.django_db
+def test_per_page_too_large(client: Client, multiple_published_digests):
+    response = client.get(f'{DIGESTS_URL}?per-page=10&page=1')
+    assert response.status_code == 200
```
Comment (thewilkybarkid) [inline app/digests/tests/test_digest_api.py]: I'd expect a `400`.

File: app/digests/tests/test_digest_api.py [hunk 4]
```diff
@@ -134,3 +135,43 @@ def test_can_filter_on_digest_stage(stage: str,
...
+    assert len(response.data['items']) == len(multiple_published_digests)
+
+@pytest.mark.django_db
+def test_page_too_large(client: Client, multiple_published_digests):
+    all_digests = len(multiple_published_digests)
+    response = client.get(f'{DIGESTS_URL}?per-page={all_digests}&page=2')
+    assert response.status_code == 404
+
+@patch('digests.pagination.DigestPagination.max_page_size', 5)
```
Comment (thewilkybarkid) [inline app/digests/tests/test_digest_api.py]: Erm...

Comment (giorgiosironi) [reply app/digests/tests/test_digest_api.py]: Better than inserting 101 items in the database for a single test?

File: app/digests/tests/test_digest_api.py (hunk 1 above)
Comment (giorgiosironi) [reply app/digests/tests/test_digest_api.py]: https://github.com/elifesciences/api-raml/blob/develop/src/api.raml#L150 specifies a 404 but unclear for which situations, which are 400 and which are 404?

File: app/digests/tests/test_digest_api.py (hunk 4 above)
Comment (thewilkybarkid) [reply app/digests/tests/test_digest_api.py]: Shouldn't need any items...

File: app/digests/tests/test_digest_api.py (hunk 1 above)
Comment (thewilkybarkid) [reply app/digests/tests/test_digest_api.py]: That's not an exhaustive list. Not match the spec (https://github.com/elifesciences/api-raml/blob/3966aeb1666f8766e104ae17561f6e6f966e2a54/src/api.raml#L122-L129) means a `400`.

File: app/digests/tests/test_digest_api.py (hunk 4 above)
Comment (giorgiosironi) [reply app/digests/tests/test_digest_api.py]: We can probably pre-validate some of these in the `api` module, to cater for various 400 errors.

File: app/digests/tests/test_digest_api.py [hunk 5]
```diff
@@ -134,3 +135,54 @@ def test_can_filter_on_digest_stage(stage: str,
...
+    last_page = 4
+    total = sum([_count_items(page) for page in range(1, last_page)]) + _count_items(page=last_page)
+
+    assert total == len(multiple_published_digests)
+
+@pytest.mark.django_db
+def test_page_invalid(client: Client):
+    response = client.get(f'{DIGESTS_URL}?page=the_second_one_please')
+    assert response.status_code == 400
+    assert response.data == {'title': '`page` parameter is invalid'}
```
Comment (giorgiosironi) [inline app/digests/tests/test_digest_api.py]: avoiding reflecting any input, but everything is logged

**Comment by giorgiosironi** [comment]
https://alfred.elifesciences.org/job/test/job/test-digests/34/console shows a problem in smoke tests, have to debug this later.

**Pull request merged**